File: mantis-free-react-admin-template/src/data/get_greed_fear_index.py
import requests

FNG_TODAY_API_URL = "https://api.alternative.me/fng/?limit=1&format=json"

def get_today_grade_and_index():
    response = requests.get(FNG_TODAY_API_URL)
    today_data = response.json().get("data")[0]
    
    today_index = int(today_data['value'])
    today_grade = today_data['value_classification'].title()
    
    return today_grade, today_index

if __name__ == "__main__":
    grade, index = get_today_grade_and_index()
    print(f"Grade: {grade}")
    print(f"Index: {index}")

File: mantis-free-react-admin-template/src/data/get_greed_fear_index.js
const FNG_TODAY_API_URL = "https://api.alternative.me/fng/?limit=1&format=json";

const get_today_grade_and_index = async () => {
    try {
        const response = await fetch(FNG_TODAY_API_URL);
        const data = await response.json();
        const today_data = data.data[0];
        
        const today_index = parseInt(today_data.value);
        const today_grade = today_data.value_classification.charAt(0).toUpperCase() + 
                          today_data.value_classification.slice(1).toLowerCase();
        
        return [today_grade, today_index];
    } catch (error) {
        console.error('Error fetching Fear & Greed Index:', error);
        return ['Error', 0];
    }
};

export default get_today_grade_and_index; 